### PR TITLE
Add a console command to manage testing instructions as segment files

### LINKED ISSUE #7059: Add console commands to manage testing instructions
Similar to https://github.com/woocommerce/woocommerce-admin/issues/6837,

Add `add` and `write` commands to add testing instructions files and write them into `TESTING-INSTRUCTIONS.md` when ready.

New workflow would be:

1. Work on an issue and make a PR
2. Use `npm run tsi add`. The command asks to name the testing instruction and PR #. It also creates a markdown file in `/testing-instructions
3. Open the markdown file and add testing instructions as usual.
4. Commit the markdown file. 

When we're ready to release, the release lead executes `npm run tsi write :version` to merge the individual testing instruction files into `TESTING-INSTRUCTIONS.md`

### PR DESCRIPTION
**Pull request by moon0326**
Fixes #7059 

This PR adds a new console command to manage testing instructions as segment files in order to improve testing instruction merge conflicts we've been experiencing. 

It introduces a new workflow for adding testing instructions.

1. Open a PR for the issue to create a PR # as usual.
2. Go back to the local repository and run `npm run tsi -- add`. The command asks for the name and PR # to create a MD file in `/testing-instructions` directory. You can optionally use `--pr` option to pull the testing instructions automatically. `npm run tsi -- add --pr=:number`
3. Open the created MD file and fill out the testing instructions.
4. Commit the file.

Once we're ready to release a new version,

1. Run `npm run tsi -- write :release-version` to merge individual MD files in `/testing-instructions` into `TESTING-INSTRUCTIONS.md`

### Screenshots
Screen Shot 2021-05-25 at 7 07 54 AM
Screen Shot 2021-05-25 at 7 08 38 AM

### CODE REVIEW

File: bin/tsi.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
```
Comment (louwie17) [inline bin/tsi.php:1]: I think we could probably rename this file to `testing-instructions`, `tsi` is kinda confusing.

File: composer.json [hunk 2]
```diff
@@ -15,7 +15,8 @@
 	"require-dev": {
 		"woocommerce/woocommerce-sniffs": "0.1.0",
 		"suin/phpcs-psr4-sniff": "^2.2",
-		"bamarni/composer-bin-plugin": "^1.4"
+		"bamarni/composer-bin-plugin": "^1.4",
+		"symfony/console": "^3.4 | ^5.2"
```
Comment (louwie17) [inline composer.json:19]: Is the ` | ^5.2` expected? It didn't install for me automattically when doing `composer install`, I end up running `composer require symfony/console`.

File: package.json [hunk 3]
```diff
@@ -76,7 +76,8 @@
 		"changelog": "node ./bin/changelog --changelogSrcType='ZENHUB_RELEASE'",
 		"bump-version": "npm run -s install-if-deps-outdated && php ./bin/update-version.php",
 		"wp-env-mysql-port": "node ./docker/wc-admin-wp-env/mysql-port.js",
-		"create-hook-reference": "node ./bin/hook-reference/index.js"
+		"create-hook-reference": "node ./bin/hook-reference/index.js",
+		"tsi": "php ./bin/tsi.php"
```
Comment (louwie17) [inline package.json:80]: I think we could use `testing-instructions` versus `tsi`, although `tsi` is shorter it doesn't reflect its purpose clearly.

**Review by louwie17 — COMMENTED** [review]
@moon0326 thanks for working on this :) It did test well aside from the initial `composer install` (mentioned below).
I was curious if after any of the changes in the changelogger if we could remove some of the helper files here as well? Or will they all be necessary given we are doing it manually?

File: composer.json (hunk 2 above)
Comment (moon0326) [reply composer.json:19]: Interesting. I actually ran `composer require symfony/console` and the file was updated automatically. I will double check.

File: package.json (hunk 3 above)
Comment (moon0326) [reply package.json:80]: Cool :) Yeah `tsi` is a bit confusing. I was just trying to use something that's shorter than `testing-instructions`.

**Comment by moon0326** [comment]
@louwie17 Thank you for the review 👍 This PR doesn't actually use Jetpack Changelogger. I think this is the minimum setup we need to have.

**Comment by joshuatf** [comment]
@moon0326 Just checking in to see if this is still being worked on or if we should close it for now.

**Comment by moon0326** [comment]
> @moon0326 Just checking in to see if this is still being worked on, needs a review, or if we should close it for now.

Hey @joshuatf I would like to continue working on this in the next cooldown period (next week). Let's keep it open 👍

**Comment by moon0326** [comment]
Closing this in favor of https://github.com/woocommerce/woocommerce-admin/issues/7481 for now.

**Pull request closed without merging**